PORTING.md

```diff
@@ -0,0 +1,354 @@
+# Porting to xpcc
+
+## What xpcc supports
+
+xpcc contains extensive support for AVR and STM32 devices, specifically:
+
+- AT90, ATtiny, ATmega devices
+- STM32 F0, F1, F3, F4, F7 devices.
+
+If your device falls into any of those, you can just use the specific device
+identifier in your `project.cfg` and it'll very likely just work out-of-the-box.
+
+If you want to add a development board for an existing device, have a look at [`xpcc/architecture/platform/board`](https://github.com/roboterclubaachen/xpcc/tree/develop/src/xpcc/architecture/platform/board).
+
+## Porting a new device from AVR or STM32
+
+It may be as easy as adding a new device file, but if it's a brand new device, the device headers also need to be updated. When porting a new series (for example the STM32L4), a lot of code can be reused, but there may be very important differences between the two that are not immediately obvious to someone who isn't that familiar with STM32 devices.
+
+We recommend writing us an email, or opening up an issue on GitHub.
```

Comment:
here I would also like to include a link to the mailing list as well as to the github issue page